test/langtools/tools/javac/valhalla/value-objects/PreloadAttributeTest.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8280164
+ * @summary Check emission of Preload attribute
+ * @modules jdk.jdeps/com.sun.tools.classfile
+ * @compile MultiValues.java
+ * @compile -g PreloadAttributeTest.java
+ * @run main PreloadAttributeTest
+ */
+
+import com.sun.tools.classfile.*;
+import com.sun.tools.classfile.ConstantPool.CONSTANT_Class_info;
+
+public class PreloadAttributeTest {
+
+    static final value class X {
+        final V1 [] v1 = null; // field descriptor
+        V2[] foo() {  // method descriptor encoding value type
+            return null;
+        }
+        void foo(V3 v3) { // method descriptor encoding value type
+        }
+        void foo(int x) {
+            V4 [] v4 = null; // local variable.
+        }
+        void goo() {
+            V5 [] v5 = null;
+            if (v5 == null) {
+               V6 [] v61 = null;  // stack map table.
+            } else {
+               V5 [] v52 = null;
```

Comment:
A note: V6 has no mention in the source file anywhere (local variable table, stack map table ...) and so doesn't feature in Preload either. The test should be updated to reflect that - it gives the impression that V6 is expected in the Preloads.